File: app/src/test/java/igrek/songbook/dagger/DaggerInjectionTest.java
package igrek.songbook.dagger;

import android.app.Activity;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;

import javax.inject.Inject;

import igrek.songbook.activity.MainActivity;
import igrek.songbook.dagger.base.DaggerTestComponent;
import igrek.songbook.dagger.base.TestComponent;
import igrek.songbook.dagger.base.TestModule;

import static org.assertj.core.api.Java6Assertions.assertThat;

@RunWith(RobolectricTestRunner.class)
public class DaggerInjectionTest {
	
	@Inject
	Activity activity;
	
	@Before
	public void setUp() {
		MainActivity activity = Robolectric.setupActivity(MainActivity.class);
		
		TestComponent component = DaggerTestComponent.builder()
				.factoryModule(new TestModule(activity))
				.build();
		
		DaggerIoc.setFactoryComponent(component);
		
		component.inject(this);
	}
	
	@Test
	public void testActivityInjection() {
		assertThat(activity).isNotNull();
		System.out.println("injected activity: " + activity.toString());
	}
	
}

File: app/src/test/java/igrek/songbook/dagger/DaggerInjectionTest.kt
package igrek.songbook.dagger

import android.app.Activity
import assertk.assertThat
import assertk.assertions.isNotNull
import igrek.songbook.activity.MainActivity
import igrek.songbook.dagger.base.DaggerTestComponent
import igrek.songbook.dagger.base.TestModule
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito
import javax.inject.Inject

class DaggerInjectionTest {

    @Inject
    lateinit var activity: Activity

    @Before
    fun setUp() {
        val activity = Mockito.mock(MainActivity::class.java)

        val component = DaggerTestComponent.builder()
                .factoryModule(TestModule(activity))
                .build()

        DaggerIoc.factoryComponent = component

        component.inject(this)
    }

    @Test
    fun testActivityInjection() {
        assertThat(activity).isNotNull()
        println("injected activity: $activity")
    }

}
